### PR TITLE
Feature/check in event

### PR DESCRIPTION
**Pull request by kaupunki-apina**
- Checking into events
- Endpoint /events may filter results by city
- PostgreSQL rather than .json driven event persistence

### CODE REVIEW

File: src/core/action-core.js [hunk 1]
```diff
@@ -31,11 +32,19 @@ function createAction(action) {
...
+          createFeedItem(action, trx);
         }
 
         return Promise.resolve();
-      });
+    })
+  })
+  .then(() => undefined)
+  .catch(err => {
+    if (err.constraint === 'only_one_check_in_per_event') {
```
Comment (miro) [inline src/core/action-core.js]: this string COULD be defined somewhere as a constant, and you could use that definition in here & in the index creation as well... but not a blocker

File: migrations/20170310092939_add_events_table.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+
+exports.up = function(knex, Promise) {
+  return knex.schema.createTable('events', table => {
+    table.string('id').primary().index();
```
Comment (tomi) [inline migrations/20170310092939_add_events_table.js]: It's usually a good idea to have a numeric surrogate primary key field as it for example makes joins faster. If a field is needed to identify the row from application code, a separate field can be used. So summa summarum:

* Change the primary key to `table.increments('id').primary().index();`
* Add a separate `code` field `table.string('code').notNullable().unique().index();`

File: seeds/events.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+exports.seed = function(knex, Promise) {
+  return knex('cities')
+    .select('id')
+    .select('name')
+    .then(cities => cities.filter(city => _fileExists(city.name)))
+    .then(cities => {
+      return BPromise.map(cities, city => {
+        return BPromise.map(_readEvents(city), (event, index) => {
+          return util.insertOrUpdate(knex, 'events', {
+            id: `${ city.name }_${ index }`,
```
Comment (tomi) [inline seeds/events.js]: This should be a `code` field. Update should be done using it in `WHERE` condition

File: src/core/event-core.js [hunk 4]
```diff
@@ -1,36 +1,102 @@
...
 
-  event.attendingCount = attendingCount;
+  return whereClauses;
+}
+
+// Checks if checking in with the given parameters would be feasable.
+// DOES NOT check if user has already checked in, thus the result
+// is not a guarantee of a successfull check in.
+function isValidCheckIn(action) {
+  return knex('events').select(['*']).where('id', '=', action.eventId)
```
Comment (tomi) [inline src/core/event-core.js]: What's the difference between `select(['*'])` and `select('*')`?

File: src/core/event-core.js [hunk 5]
```diff
@@ -1,36 +1,102 @@
...
+function _eventOnGoing(event) {
+  return moment().utc().isBetween(
+    moment(event.start_time).utc(),
+    moment(event.end_time).utc()
+  );
+}
+
+function _userInVicinity(actionLocation, eventLocation, eventRadius) {
+  let earthRadius = 6372.8;
+  let distanceToEvent = equirectangularDistance(
```
Comment (tomi) [inline src/core/event-core.js]: These could be `const` instead of `let` as they don't change

File: src/util/fb.js [hunk 6]
```diff
@@ -29,17 +30,20 @@ const state = {
...
-      setInterval(_updateFromFacebook, REFRESH_INTERVAL);
-
-      // Execute update immediately
-      _updateFromFacebook();
-    });
+  knex('events')
+    .select('fb_event_id AS facebookId')
+    .whereNotNull('fb_event_id')
+    .then(events => {
+      console.log(events);
```
Comment (tomi) [inline src/util/fb.js]: console.log

**Pull request merged**